cpp/ScanChain_Hgg.C

```diff
@@ -409,6 +438,66 @@ int ScanChain_Hgg(TChain *ch, double genEventSumw, TString year, TString process
 
       if (dijets[0].p4.M()<50) continue;
 
+      if (isMC){
+        LeadPhoton_genPartFlav = int(selectedDiPhoton.leadPho.genPartFlav());
+        SubleadPhoton_genPartFlav = int(selectedDiPhoton.subleadPho.genPartFlav());
+        TLorentzVector GenHiggs, GenX, GenY;
+        GenParts genparts = getGenParts();
+        vector<TLorentzVector> gen_child_xyh, gen_child_ygg, gen_child_hbb;
+        for (int igenpart=0; igenpart<genparts.size(); igenpart++)
+          {
+            if (genparts[igenpart].isxyh()) {
+              GenX = genparts[igenpart].mother_p4(); 
+              gen_child_xyh.push_back(genparts[igenpart].p4());
+              GenX_pt = GenX.Pt();
+              GenX_eta = GenX.Eta();
+              GenX_phi = GenX.Phi();
+              GenX_mass = GenX.M();
+            }
+            if (genparts[igenpart].isygg()) {
+              GenY = genparts[igenpart].mother_p4(); 
+              gen_child_ygg.push_back(genparts[igenpart].p4());
+              GenY_pt = GenY.Pt();
+              GenY_eta = GenY.Eta();
+              GenY_phi = GenY.Phi();
+              GenY_mass = GenY.M();
+            }
+            if (genparts[igenpart].ishbb()) {
+              GenHiggs = genparts[igenpart].mother_p4();
+              gen_child_hbb.push_back(genparts[igenpart].p4());
+              GenHiggs_pt = GenHiggs.Pt();
+              GenHiggs_eta = GenHiggs.Eta();
+              GenHiggs_phi = GenHiggs.Phi();
+              GenHiggs_mass = GenHiggs.M();
+            }
+          }
+        TLorentzVector sort_GenPart;
+//          if (gen_child_xyh[0].Pt()<gen_child_xyh[1].Pt()) {sort_GenPart = gen_child_xyh[0]; gen_child_xyh[0]= gen_child_xyh[1]; gen_child_xyh[1] = sort_GenPart;}
+//          if (gen_child_ygg[0].Pt()<gen_child_ygg[1].Pt()) {sort_GenPart = gen_child_ygg[0]; gen_child_ygg[0]= gen_child_ygg[1]; gen_child_ygg[1] = sort_GenPart;}
+        if (!abs(GenX_pt+999)<0.0001){
```

Comment:
For this `if` and the next ones: what is their purpose? What's the situation we are trying to avoid?

Reply:
Because the default value of GenX_pt is -999, for example we are looping over ZZ MC, there isn't a GenX in the MC, so GenX_pt will keep the default value. And if that's the case, we will hit problems if we want to get the gen_child